python/lsst/meas/base/diaCalculationPlugins.py

```diff
@@ -79,6 +86,343 @@ def wrapperCatchWarnings(*args, **kwargs):
+                The number of detections.
+            period_max_grid : `float`
+                The maximum period in the grid.
+            zmax : `float`
+                The maximum power in the grid.
+
+            Returns:
+            -------
+            fap_estimate : `float`
+                The False-Alarm probability Baluev approximation.
+
+            References:
+            ----------
+            .. [1] Baluev, R. V. 2008, MNRAS, 385, 1279
+            .. [2] Süveges, M., Guy, L.P., Eyer, L., et al. 2015, MNRAS, 450, 2052
+            """
+            gam_ratio = (gamma((N-1)/2))/(gamma((N-2)/2))
+            fu = 1/period_max_grid
+            return gam_ratio * np.sqrt(4*np.pi*statistics.variance(time)
+                                       ) * fu * (1-zmax)**((N-4)/2) * np.sqrt(zmax)
+
+        def _calculate_period_multi(df, min_detections=9, oversampling_factor=5, nyquist_factor=100):
+            """ Calculate the multi-band Lomb-Scargle periodogram.
+
+            Parameters:
+            ----------
+            df : `pandas.DataFrame`
+                The input DataFrame.
+            min_detections : `int`, optional
+                The minimum number of detections. Default is 130.
+            oversampling_factor : `int`, optional
+                The oversampling factor for frequency grid. Default is 5.
+            nyquist_factor : `int`, optional
+                The Nyquist factor for frequency grid. Default is 100.
+
+            Returns:
+            -------
+            pd_tab : `pandas.Series`
+                The output DataFrame with the Lomb-Scargle parameters.
+            """
+            tmpDf = df[~np.logical_or(np.isnan(df["psfFlux"]),
+                                      np.isnan(df["midpointMjdTai"]))]
+
+            if (len(tmpDf)) < min_detections:
+                return pd.Series({periodCol: np.nan,
+                                  powerCol: np.nan,
+                                  fapCol: np.nan,
+                                  ampCol: np.nan,
+                                  phaseCol: np.nan})
+
+            time = tmpDf["midpointMjdTai"].to_numpy()
+            flux = tmpDf["psfFlux"].to_numpy()
+            flux_err = tmpDf['psfFluxErr'].to_numpy()
+            bands = tmpDf['band'].to_numpy()
+
+            # TODO: currently assuming 1-1 base-band model.
+            lsp = lspm(time, flux, bands, dy=flux_err,
+                       nterms_base=1, nterms_band=1)
+
+            f_grid = compute_optimized_periodogram_grid(
+                time, oversampling_factor=oversampling_factor, nyquist_factor=nyquist_factor)
+            period = 1/f_grid
+            power = lsp.power(f_grid)  # compute power
+
+            # Calculate the False-Alarm probability Baluev approximation
+            fap_estimate = calculate_baluev_fap(
+                time, len(time), period[np.argmax(power)], np.max(power))
+
+            def generate_lsp_params(lsp_model, fbest, bands):
```

Comment:
Oh, this is a closure of a closure? I think this is getting a bit too nested. Why not make all of these local functions private static class methods (with prepended `_`), since I'm not sure you actually use them as closures?

Reply:
I have moved the functions as @staticmethod to the class.